### PR TITLE
WIP: Release v 0.0.1

### PR DESCRIPTION
**Pull request by nbey**
Implement connector to sync user accounts with Looker

- Sync User Accounts
  - [x] User Roles
  - [x] User Groups
  - Custom Attributes
    - [x] School
    - [x] StudentID

- [ ] SAML2 SSO Implementation ?

- [ ] SSO Embedded Pages ?

### CODE REVIEW

**Comment by themightychris** [comment]
@nbey what did you need to customize about the synchronizeComplete template?

**Comment by nbey** [comment]
> @nbey what did you need to customize about the synchronizeComplete template?

@themightychris I may have just copied that template after getting an error from `createJob.tpl` not existing. I'll do a diff and see if there are any differences, and get rid of it if not.

**Comment by nbey** [comment]
So far, I've set up the following workflows for customizing roles/groups:

- They can be set by the given users' Class, or by the given users' Account Level.

For Custom Attributes, they can currently be set by users' Class.

examples:
- set user role via config:
```
Connector::$rolesByAccountType = [
    User::class => [2,4,6] // array of roleIds to set for non-students,
    Student::class => [1,3,5] // array of roleIds to set for students
];
```
- set user groups via config
```
Connector::$groupsByAccountLevel = [
    "Administrator" => [2,4,6] // array of groupIds to set for Administrators
];
```
- set custom attributes via config
```
Connector::$customAttributesByType = [
  Student::class => [
    'student_id' => [
      'getter' => 'StudentNumber'
    ],
    'school' => [
      'value' => Slate::$schoolName
    ]
  ]
];
```

**Comment by nbey** [comment]
@themightychris 
I've tested the sync on `cbl-develop` with an account and have successfully synced each of these values.

I think one thing we may need to consider is generating credentials for the user (email, saml, api, etc).

File: php-classes/Slate/Connectors/Looker/Connector.php [hunk 1]
```diff
@@ -0,0 +1,587 @@
...
+        }
+
+        if (empty($Job->Config['clientSecret'])) {
+            return static::throwError('Cannot execute job, clientSecret not provided');
+        }
+
+
+        // configure API wrapper
+        LookerAPI::$clientId = $Job->Config['clientId'];
+        LookerAPI::$clientSecret = $Job->Config['clientSecret'];
```
Comment (themightychris) [inline php-classes/Slate/Connectors/Looker/Connector.php]: remove in favor of calling `login` and passing job creds

File: php-classes/Slate/Connectors/Looker/Connector.php [hunk 2]
```diff
@@ -0,0 +1,587 @@
...
+                'Created Looker user for {slateUsername}',
+                    [
+                        'slateUsername' => $User->Username,
+                    'lookerResponse' => $pretend ? '(pretend-mode)' : $lookerResponse
+                    ]
+                );
+
+            // sync groups
+            try {
+            $groupSyncResult = static::syncUserGroups($User, [], $logger, $pretend);
```
Comment (themightychris) [inline php-classes/Slate/Connectors/Looker/Connector.php:265]: indent

File: php-classes/Slate/Connectors/Looker/Connector.php [hunk 3]
```diff
@@ -0,0 +1,587 @@
...
+                    $e->getContext()
+                );
+            }
+
+            return new SyncResult(
+                SyncResult::STATUS_CREATED,
+                'Created Looker user for {slateUsername}, saved mapping to new Looker user #{lookerUserId}',
+                    [
+                        'slateUsername' => $User->Username,
+                    'lookerUserId' => $pretend ? '(pretend-mode)' : $lookerResponse['id'],
```
Comment (themightychris) [inline php-classes/Slate/Connectors/Looker/Connector.php:296]: indent

**Comment by themightychris** [comment]
@nbey so what we need next is to start exploring a meta-SAML integration. Spin up an empty site based on skeleton-v2 and pull in this layer: https://github.com/JarvusInnovations/emergence-saml2

In your first pass just get login working with some test users inside your site

**Comment by themightychris** [comment]
@nbey here is an example of extending the SAML connector to override some of the data sent back to the SP: https://github.com/JarvusInnovations/emergence-slack/blob/master/php-classes/Emergence/Slack/Connector.php#L155

**Comment by nbey** [comment]
I updated the login method, and looked for those indention issues but they looked to have been resolved already.

**Pull request merged**